### ISSUE TITLE
How to modify an already merged operator?

### ISSUE THREAD

**Issue by benjaminjb** [redhat-openshift-ecosystem/certified-operators#1702]
The readme states "Pull request can't modify already merged operators"; so what is the protocol for updating an already merged operator bundle?

After we merged [this PR](https://github.com/redhat-openshift-ecosystem/certified-operators/pull/1685), we discovered a problem with the OperatorHub installer for OCP v4.8.

(For the curious, it's a problem that this PR addresses: https://github.com/operator-framework/operator-registry/commit/8720e3daeac4084753bb26d210b0fd6fce1432c5 -- our bundle is too big and v4.8 has no gzip option.)

Now we want to update all our 5.3.0 bundles so that the minimum OCP is 4.9. How do we go about updating this already merged operator bundle? Is there a set protocol that we can do or do we require different permissions to, say, delete the operator 5.3.0 bundle and re-add it with the appropriate changes?

(Duplicate of https://github.com/k8s-operatorhub/community-operators/issues/2092 since this affects both repos, in case there's a solution that covers both.)

**Comment by jsm84** [comment]
@benjaminjb  We're quickly approaching EOY shutdown, but the usual method of redacting a release is to version bump and push a new bundle that `skips` the previous bundles.

The issue here is that your versions 5.3.0+ aren't compatible with OCP 4.8, so the only way to fix the v4.8 index is to manually remove the entire operator from the index, and then incrementally add back 5.1.0, then 5.1.2, 5.1.3 and 5.2.0 (as that's the entire update graph that I can see).

Then, I would suggest pushing a fixed version of the 5.3.0 bundle with the com.redhat.openshift.versions="v4.9" label in annotations.yaml, versioned as 5.3.1.

I only have developers on hand who can sign the 4.8 index today, and then we're going dark until January.

There is one major roadblock, though: I can't currently pull the digest for the 5.1.0 bundle, and I ran into this while troubleshooting #367. The rest of the bundle versions appear to be published, however.

**Comment by jsm84** [comment]
@benjaminjb I can resolve this for certified-operators. I'm able to pull all bundle releases for certified-operators, and you haven't yet pushed v5.3.0 to redhat-marketplace yet, so this isn't affected there.

I'll post once this is fixed. You will end up with v5.2.0 and earlier for OpenShift v4.8, with v5.3.0 and earlier on all later versions.

**Comment by jsm84** [comment]
@benjaminjb I've effectively removed v5.3.0 from the OpenShift v4.8 index.

Moving forward, I do suggest my earlier suggestion, as doing this should prevent v5.3.0 from publishing on v4.8 in the future.

1. Create a new bundle version 5.3.1, otherwise identical to 5.3.0 (except the version and name within the CSV).
2. Ensure that version 5.3.1 has `com.redhat.openshift.versions="v4.9"` reflected in `metadata/annotations.yaml`
3. Set this to "skip" version 5.3.0

You can do this for both certified-operators and redhat-marketplace, however I do expect to run into the same issue as before with redhat-marketplace (where I can't pull the v5.1.0 bundle). This is likely why the index sync didn't occur properly for RHM, as the base version was missing or unpublished.

**Comment by benjaminjb** [comment]
Hi @jsm84, first, thanks for working on this.

Second, I've made a PR with the updated version (4.8->4.9) as I'm hoping that we can move forward with that, but if that's a problem, please let us know.

(And I'll update the unmerged bundles so they match.)

**Comment by benjaminjb** [comment]
(Or do we need to merge a PR that removes the files and then open another PR with the corrected files?)

**Comment by benjaminjb** [comment]
Hey @jsm84, just wanted to double-check on this: is there a best practice for this situation?

**Comment by jsm84** [comment]
@benjaminjb Unfortunately, you cannot modify a release that has already been published.

The [PR](https://github.com/redhat-openshift-ecosystem/certified-operators/pull/1703) cannot merge because it's modifying files/directories which already exist in the repo.

The way forward, without me having to completely remove your operator from the certified index for OCP 4.9 thru 4.12, and sequentially restore each point release, is to push 5.3.1 as a new bundle set to skip the 5.3.0 release.

**Comment by jsm84** [comment]
@benjaminjb You could also just address this at the time of the next release (5.4.0?) with the changes (skip 5.3.0, minimum OCP version 4.9) mentioned previously.

**Comment by benjaminjb** [comment]
Thanks for the help and just want to clarify: can we merge a PR that removes the 5.3.0 bundle?

There's a little precedent here that I'm trying to check out:

https://github.com/redhat-openshift-ecosystem/certified-operators/pull/775 : 5.1.0 bundle was completely removed
https://github.com/redhat-openshift-ecosystem/certified-operators/pull/776:  5.1.0 bundle was re-added with fix

Do you know if there was some special operation that Tony did behind the scenes with those PRs?

**Comment by jfrancin** [comment]
Yes there absolutely was a special operation that Tony did. This was back when we had first implemented this process. He removed 5.1.0 via deleting the files, and then someone on your side created a new PR with a fixed version of v5.1.0.  This was only possible because, at the time, v5.1.0 was at the tip of the OLM skips/replaces chain. It was also before we modified the rules for manual work: due to the existing workload on engineers , we cannot spare the time to perform the manual steps required to work around the documented process - which is to use 'skips' or 'skipRange' directives to skip over bad versions.
